src/concurrency/timestamp_ordering_transaction_manager.cpp

```diff
@@ -624,6 +623,15 @@ ResultType TimestampOrderingTransactionManager::CommitTransaction(
   //// handle other isolation levels
   //////////////////////////////////////////////////////////
 
+  auto &rw_set = current_txn->GetReadWriteSet();
+
+  // if no modifying queries, treat as read-only
+  if (rw_set.empty()) {
```

Comment:
Based on the implementation of the is_written_ flag in #1401 could we also check the is_written_ flag here? That would tell us if the RWSet is just reads.

Reply:
@mbutrovich I've tried changing it to use is_written_ instead of empty(), but I keep breaking on [multi_granularity_access_test](https://github.com/cmu-db/peloton/blob/master/test/concurrency/multi_granularity_access_test.cpp#L149). Did you and/or @pervazea say something about SCAN being broken earlier?